docs/upgrade/xenial_upgrade_in_place.rst

```diff
@@ -0,0 +1,317 @@
+recommend performing the following tests yourself to identify potential issues
+specific to your system configuration.
+
+Validate the kernel version
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ensure you are logged out, and then type the commands ``ssh app uname -r`` and
+``ssh mon uname -r`` in your terminal window.
+
+The output for both commands should be ``4.4.167-grsec``, which indicates that
+the latest available kernel for SecureDrop is installed on your *Application
+Server* and your *Monitor Server*.
+
+Validate the application version
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To confirm that you are running SecureDrop 0.12.0 for Xenial, on the Tails
+desktop, you should find a shortcut called **SecureDrop Source Interface**.
+Double-click it to launch the Tor browser.
+
+After the *Source Interface* loads, add the path ``/metadata`` to the URL in
+your address bar. If your *Source Interface* can be found at
+``examplenot4real.onion``, then the address you should visit is
+``examplenot4real.onion/metadata``. That page should show you key/value pairs,
+including ``0.12.0`` for ``sd_version`` and ``16.04`` for ``server_os``.
+
+End-to-end test
+^^^^^^^^^^^^^^^
+
+We recommend an end-to-end test of document submission, reply and decryption.
+First, confirm that you can log into the *Journalist Interface*. On the Tails
+desktop, you should find a shortcut called **SecureDrop Journalist Interface**.
+Double-click it to launch the Tor browser.
+
+Once the page has finished loading, sign in using your SecureDrop login
+credentials. Confirm that you can view the list of submissions as expected.
+
+Keep the browser window open, and launch the **SecureDrop Source Interface**
+using its shortcut on the Tails desktop. The *Source Interface* should load in
+another browser tab.
+
+Once the page has finished loading, click **Submit Documents**. On the subsequent
+page, click **Submit Documents** again (you may want to write down your codename
+in case you need it for further testing).On the following screen, choose a
+simple file to upload, and enter a message to go along with it, then press
+**Submit**.
+
+Switch to the tab with the *Journalist Interface*, reload it, and confirm that
+you can see your new submission. Write a reply, and switch back to the
+*Source Interface*. Reload it, and confirm that you can see the reply.
+
+Now, from the *Journalist Interface*, download the submission you just made.
+Copy it to your *Transfer Device* and boot into your *Secure Viewing Station*.
+Confirm that you can open the encrypted document.
+
+Just in case you picked the wrong submission, we strongly recommend following
+standard precautions, e.g., do not open the document directly from the *Transfer
+Device* but copy it onto the *Secure Viewing Station* first.
+
+Contact us
+----------
+If you have questions or comments regarding this process, or if you
+encounter any issues, you can always contact us by the following means:
```

Comment:
In your experience, how much instance-specific data are in the error messages when upgrades fail? Enough to recommend they be mindful of org-specific data in public forums (forum, github)